pkg/target/kusttarget.go

```diff
@@ -275,7 +274,7 @@ func (kt *KustTarget) runGenerators(
 		if err != nil {
 			return err
 		}
-		err = ra.AppendAll(resMap)
```

Comment:
I'm disappointed (in my past self likely) that this change doesn't cause a test to fail.  :(

On generation, this will merge (allow id collisions) instead of restricting to only adding new objects.
OK, probably more upside than downside, and its hard to imagine someone depending on the current behavior to avoid errors.  If we get complaints, we can add a flag.   one of many reasons why we more data about what features are being used.

OK, yes please - can you add some regression tests please?

Reply:
> I'm disappointed (in my past self likely) that this change doesn't cause a test to fail. :(
> 
> On generation, this will merge (allow id collisions) instead of restricting to only adding new objects.

The default behavior for `AbsorbAll` if a resource doesn't specify merge behavior via the generator options is to use `Append`. So would only be an issue if a go plugin author (only plugins that currently have any mechanism for setting generator options) is explicitly setting `BehaviorReplace` or `BehaviorMerge` when calling the resource factory methods and then depending on the fact that the behavior is ignored... feels rather unlikely. 😄

> OK, yes please - can you add some regression tests please?

Absolutely.